src/org/usfirst/frc/team2585/systems/ShooterSystem.java

```diff
@@ -0,0 +1,101 @@
+package org.usfirst.frc.team2585.systems;
+
+import org.impact2585.lib2585.Toggler;
+import org.usfirst.frc.team2585.Environment;
+import org.usfirst.frc.team2585.RobotMap;
+import org.usfirst.frc.team2585.input.InputMethod;
+
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.Victor;
+
+/**
+ * This system controls the ball shooter of the robot
+ */
+public class ShooterSystem implements RobotSystem, Runnable {
+	private InputMethod input;
+	
+	private SpeedController agitator;
+	private SpeedController shooter;
+	
+	private double prevShooterSpeed;
+	
+	public static final double RAMP = 0.6;
+	
+	private Toggler shooterToggler;
+	
+	/**
+	 * Constructor for the shooter system that initializes the togglers
+	 */
+	public ShooterSystem() {
+		shooterToggler = new Toggler(false);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.usfirst.frc.team2585.systems.Initializable#init(org.usfirst.frc.team2585.Environment)
+	 */
+	@Override
+	public void init(Environment environ) {
+		agitator = new Victor(RobotMap.AGITATOR);
+		input = environ.getInput();
+	}
+	
+	public void setInput(InputMethod newInput) {
```

Comment:
Don't forget your javadoc comments. Also you should probably make RobotSystem into an abstract class that implements Destroyable and Initializable. That way you can store the environment and inputmethod instance variables in RobotSystem and have setInput() set the input for all the chilld classes. Also the init() in RobotSystem could automatically set input from the environ so that you can just call super.init(environ) in all of the child classes' init method. Don't forget to make input protected in RobotSystem.